clang/unittests/ASTMatchers/ASTMatchersNarrowingTest.cpp

```diff
@@ -2235,6 +2235,21 @@ TEST_P(ASTMatchersTest, ArgumentCountIs_CXXConstructExpr) {
                  Constructor1Arg));
 }
 
+TEST_P(ASTMatchersTest, hasDependentName_DependentScopeDeclRefExpr) {
+  if (!GetParam().isCXX() || GetParam().hasDelayedTemplateParsing()) {
+    // FIXME: Fix this test to work with delayed template parsing.
+    return;
+  }
+
+  EXPECT_TRUE(matches("template <class T> class X : T { void f() { T::v; } };",
+                      dependentScopeDeclRefExpr(hasDependentName("v"))));
+
+  EXPECT_TRUE(
```

Comment:
It would be nice to have a test case where the dependent name refers to a function (specifically a static member function, since I think that will get a `DependentScopeDeclRefExpr` rather than a `CXXDepdendentScopeMemberExpr`).

Come to think of it, this applies for the test for `dependentScopeDeclRefExpr` itself.

Reply:
Done, thank you for suggesting it :D